src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -32,29 +34,32 @@ private int getRandomFigure() {
     }
 
     private Circle createCircle() {
-        return new Circle(colorSupplier.getRandomColor(), random.nextInt(defalteSide));
+        randomSide = random.nextInt(defalteSide);
+        randomColor = colorSupplier.getRandomColor();
+        return new Circle(randomColor, randomSide);
     }
 
     private Square createSquare() {
-        return new Square(colorSupplier.getRandomColor(), random.nextInt(defalteSide));
+        randomSide = random.nextInt(defalteSide);
+        randomColor = colorSupplier.getRandomColor();
+        return new Square(randomColor, randomSide);
     }
 
     private Rectangle createRectangle() {
-        return new Rectangle(colorSupplier.getRandomColor(),
-                random.nextInt(defalteSide),
-                random.nextInt(defalteSide));
+        randomSide = random.nextInt(defalteSide);
+        randomColor = colorSupplier.getRandomColor();
+        return new Rectangle(randomColor, randomSide, randomSide);
```

Comment:
cad rectangle have different side for length and width? please also fix the same for `IsoscelesTrapezoid`